### PR TITLE
Adds I2C implementation and basic accelerometer crate.

### PR DESCRIPTION
**Pull request by tcr**
This commits the I2C interface and an accelerometer crate that can read acceleration values.

Note that API design questions should be surfaced on #29. Our goal is to get our interfaces working and tested asap, but design the ergonomics more carefully before 1.0.

### CODE REVIEW

**Comment by Dr-Emann** [comment]
`Rc<Mutex<T>>` is probably not what you want. `Rc` is not shareable across threads. If you want to lock the port to a single thread, you can use `Rc<RefCell<T>>` (`RefCell` is basically a mutex for a single thread). If you want it to be shareable across threads, you want `Arc<Mutex<T>>` instead.

**Comment by tcr** [comment]
@Dr-Emann Agreed; I'll swap that for this release.

File: README.md [hunk 1]
```diff
@@ -69,7 +71,7 @@ extern crate alloc_system;
 
 * Build the project (in release mode to optimize the size of the binary)
 ```
-cargo build --release
+cargo build --release --target=mipsel-unknown-linux-gnu
```
Comment (johnnyman727) [inline README.md:74]: Did you add this just for clarity? I don't think I needed this but maybe my setup was just different.

**Comment by johnnyman727** [comment]
I'm not experienced enough to comment on that API design as-is but I'm curious about why the `accel-mma84` driver is living within the `tessel-rust` repo. Shouldn't that be an addition to [its own repo](https://github.com/tessel/rust-accel-mma84)?

**Comment by tcr** [comment]
@johnnyman727 I think we did something like this when we started with T1; basically we had all the code in one repo to help evolve the API all at once, and broke out the modules once the API settled down. I expect that to happen here.

Also, whoops, did not notice the [Rust version of the MMA84](https://github.com/tessel/rust-accel-mma84/blob/master/src/lib.rs) code. Probably past the point of using it now though :(

File: README.md (hunk 1 above)
Comment (tcr) [reply README.md:74]: Yeah, mine required it. Guess it doesn't hurt to be specific when cross-compiling.

**Pull request merged**